index.d.ts

```diff
@@ -0,0 +1,36 @@
+import FakeXMLHttpRequest from "fake-xml-http-request";
+import { Params, QueryParams } from "route-recognizer";
+type SetupCallback = (this: Server) => void;
+interface SetupConfig {
+  forcePassthrough: boolean;
+}
+export type Config = SetupCallback | SetupConfig;
+export class Server {
+  // HTTP request verbs
+  public get: RequestHandler;
+  public put: RequestHandler;
+  public post: RequestHandler;
+  public patch: RequestHandler;
+  public delete: RequestHandler;
+  public options: RequestHandler;
+  public head: RequestHandler;
+  constructor(setup?: SetupCallback);
+
+  public shutdown(): void;
+}
+
+export type RequestHandler = (
+  urlExpression: string,
+  response: ResponseHandler
+) => void;
```

Comment:
I think there is a fourth argument here `async: boolean`

https://github.com/pretenderjs/pretender/blob/fc363b330469d20fb7b1dad4a3d7a0852b3025c4/pretender.js#L325